### PR TITLE
Handle switching of displayed items clearly, fix a bug

### PR DESCRIPTION
**Pull request by jankalfus**
I've added a showProgressBar method and modified the other methods accordingly. The idea here is that you either want to show the progress bar, the response, or the error message - not more than one of these things at once.

This makes the code much cleaner to read. It also fixes a bug, where if you do a request resulting in an error an then you make another one, both the error message and loading indicator are visible. But only the indicator should be visible.

### CODE REVIEW

File: Lesson02-GitHub-Repo-Search/T02.06-Solution-AddPolish/app/src/main/java/com/example/android/datafrominternet/MainActivity.java [hunk 1]
```diff
@@ -42,7 +42,7 @@
     private TextView mErrorMessageDisplay;
 
     // COMPLETED (20) Create a ProgressBar variable to store a reference to the ProgressBar
-    private ProgressBar mLoadingIndicator;
+    private ProgressBar mProgressBar;
```
Comment (mnemonicflow) [inline Lesson02-GitHub-Repo-Search/T02.06-Solution-AddPolish/app/src/main/java/com/example/android/datafrominternet/MainActivity.java]: What exactly did you fix in this commit besides renaming the member variable?

**Comment by mnemonicflow** [comment]
I think there are too many commits in this PR and they should be [squashed](https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/) to a single one, before it can be fully reviewed.

File: Lesson02-GitHub-Repo-Search/T02.06-Solution-AddPolish/app/src/main/java/com/example/android/datafrominternet/MainActivity.java (hunk 1 above)
Comment (jankalfus) [inline Lesson02-GitHub-Repo-Search/T02.06-Solution-AddPolish/app/src/main/java/com/example/android/datafrominternet/MainActivity.java]: Just the variable name, it was inconsistent with the rest of the changes I've made.

**Comment by jankalfus** [comment]
Done, squashed the commits. Hopefully, it's more clear now.

**Comment by mnemonicflow** [comment]
@jankalfus The PR still shows all the commits, it should say "Commits 1" at the top. You need to push with the --force flag in your fork. Although I don't think they will accept it either way, because this repository is constructed with a branch per project.
You'd have to send your PR to the respective branch associated with the changes you've made.

**Pull request closed without merging**